src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModelUpdater.java

```diff
@@ -120,12 +117,14 @@ Course getCourse(@Nullable URL courseUrl) {
       if (!projectModules.contains(module.getName())) {
         continue;
       }
-      if (!module.getVersionId().equals(localModuleIds.get(module.getName()))) {
+      if (!module.getVersionId().equals(localModuleIds.get(module.getName()))
+          // todo: a bit unsure here
+          && module.hasLocalChanges()) {
```

Comment:
is this the correct place for this check to happen? :confused:

Reply:
I would gladly delegate plugging this in for a person who will be constructing the whole thing (this feature) together :smile: